src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/AssemblyPart.cs

```diff
@@ -41,15 +41,37 @@ public AssemblyPart(Assembly assembly)
         public IEnumerable<TypeInfo> Types => Assembly.DefinedTypes;
 
         /// <inheritdoc />
-        public IReadOnlyList<CompilationLibrary> GetCompilationLibraries()
+        public IEnumerable<string> GetReferencePaths()
         {
             var dependencyContext = DependencyContext.Load(Assembly);
             if (dependencyContext != null)
             {
-                return dependencyContext.CompileLibraries;
+                var referencePaths = new List<string>();
+                var compileLibraries = dependencyContext.CompileLibraries;
+
+                try
+                {
+                    for (var i = 0; i < compileLibraries.Count; i++)
+                    {
+                        var compileLibrary = compileLibraries[i];
+                        referencePaths.AddRange(compileLibrary.ResolveReferencePaths());
+                    }
+
+                    return referencePaths;
+                }
+                catch
```

Comment:
Still seem to hit issues even w/o `preserveCompilationContext: true`. See a FNF exception with `Message` "Could not load file or assembly 'ClassLibrary, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies. The system cannot find the file specified." Oddly the exception is `thrown` when `DefaultRoslynCompilationService` asks the successfully-generated for its exported types. Seems like the compilation dependencies (and their full paths) aren't preserved for runtime.